nifi-nar-bundles/nifi-lumberjack-bundle/nifi-lumberjack-processors/src/main/java/org/apache/nifi/processors/lumberjack/ListenLumberjack.java

```diff
@@ -0,0 +1,211 @@
+        @WritesAttribute(attribute="mime.type", description="The mime.type of the content which is text/plain")
+})
+@SeeAlso(classNames = {"org.apache.nifi.processors.standard.ParseSyslog"})
+public class ListenLumberjack extends AbstractListenEventBatchingProcessor<LumberjackEvent> {
+
+    public static final PropertyDescriptor SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("SSL Context Service")
+            .description("The Controller Service to use in order to obtain an SSL Context. If this property is set, " +
+                    "messages will be received over a secure connection.")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    private volatile LumberjackEncoder lumberjackEncoder;
+    private volatile byte[] messageDemarcatorBytes; //it is only the array reference that is volatile - not the contents.
+
+
+
+    @Override
+    protected List<PropertyDescriptor> getAdditionalProperties() {
+        return Arrays.asList(MAX_CONNECTIONS, SSL_CONTEXT_SERVICE);
+    }
+
+    @Override
+    @OnScheduled
+    public void onScheduled(ProcessContext context) throws IOException {
+        super.onScheduled(context);
+        // wanted to ensure charset was already populated here
+        lumberjackEncoder = new LumberjackEncoder();
+    }
+
+    @Override
+    protected ChannelDispatcher createDispatcher(final ProcessContext context, final BlockingQueue<LumberjackEvent> events) throws IOException {
+        final EventFactory<LumberjackEvent> eventFactory = new LumberjackEventFactory();
+        final ChannelHandlerFactory<LumberjackEvent,AsyncChannelDispatcher> handlerFactory = new LumberjackSocketChannelHandlerFactory<>();
+
+        final int maxConnections = context.getProperty(MAX_CONNECTIONS).asInteger();
+        final int bufferSize = context.getProperty(RECV_BUFFER_SIZE).asDataSize(DataUnit.B).intValue();
+        final Charset charSet = Charset.forName(context.getProperty(CHARSET).getValue());
+
+        // initialize the buffer pool based on max number of connections and the buffer size
+        final BlockingQueue<ByteBuffer> bufferPool = createBufferPool(maxConnections, bufferSize);
+
+        // if an SSLContextService was provided then create an SSLContext to pass down to the dispatcher
+        SSLContext sslContext = null;
+        final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        if (sslContextService != null) {
+            sslContext = sslContextService.createSSLContext(SSLContextService.ClientAuth.REQUIRED);
+        }
+
+        // if we decide to support SSL then get the context and pass it in here
+        return new SocketChannelDispatcher<>(eventFactory, handlerFactory, bufferPool, events,
```

Comment:
@apiri 

Added cues (commit 1687169d7c932860f93bb5989c531a2d44ac06b7) and also a custome validation routine (abda1bbd7c19c46e724e682e007a7332e93fe65f)